blocks/library/block/edit-panel/index.js

```diff
@@ -21,21 +21,31 @@ class SharedBlockEditPanel extends Component {
 		super( ...arguments );
 
 		this.bindTitleRef = this.bindTitleRef.bind( this );
+		this.bindEditButtonRef = this.bindEditButtonRef.bind( this );
 		this.handleFormSubmit = this.handleFormSubmit.bind( this );
 		this.handleTitleChange = this.handleTitleChange.bind( this );
 		this.handleTitleKeyDown = this.handleTitleKeyDown.bind( this );
 	}
 
-	componentDidMount() {
-		if ( this.props.isEditing ) {
+	componentDidUpdate( prevProps ) {
+		// Select the input text only once when the form opens.
+		if ( ! prevProps.isEditing && this.props.isEditing ) {
 			this.titleRef.select();
 		}
+		// Move focus back to the Edit button after pressing Save or Cancel.
+		if ( ! this.props.isEditing && ! this.props.isSaving ) {
```

Comment:
Yeah but it wouldn't work when pressing "Cancel" 😞

Reply:
>other than the initial view (form closed)?

Correct, but we could anticipate a component update to occur and trigger focus to the edit button unexpectedly here. Maybe a plugin modifies the block title externally, triggering an update and forcing the focus? Not the best example 😄 But in general, we'd always want to make sure in this lifecycle method that something has changed between `prevProps` and `this.props`.

I think the correct version would be:

```js
const { isEditing, isSaving } = this.props;
if ( ( prevProps.isEditing && ! isEditing ) || ( prevProps.isSaving && ! isSaving ) {
	this.editButton.current.focus();
}
```

Alternatively, I wonder if we should just extract out the form into a separate component (local to the same directory, e.g. `block-panel/form.js`), and apply `withFocusReturn` to it, so that when it unmounts, we automatically shift focus back to where it was when it first mounted.